test/test.rs

```diff
@@ -100,6 +101,48 @@ cfg_if! {
     }
 }
 
+cfg_if! {
+    if #[cfg(any(target_os = "android", target_os = "linux"))] {
+        macro_rules! require_kernel_version {
```

Comment:
This macro would be a lot simpler if it used `nix::src::sys::utsname` rather than read from procfs.  That would make it portable, too.